### ISSUE TITLE
test_shear_from_matrix failes using MKL linked numpy

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#757]
### Expected behaviour

`test_transformations:test_shear_from_matrix` failes. saying that normal vectors are not orthogonal

### Actual behaviour

all tests pass

### Code to reproduce the behaviour

`./mda_nosetests test_transformation.py`

### Currently version of MDAnalysis:

git log --oneline | head -n1
5b6471d Moved .github to correct location (repo root)

**Comment by kain88-de** [comment]
I  get the same behavior when I switch to the `0.14.0` tag.

**Comment by kain88-de** [comment]
It is strange that this isn't caught by travis.

**Comment by richardjgowers** [comment]
So I don't get this failure locally either on my dev install of MDA, my numpy version is '1.10.4'.  Is conda using something more recent than that?

**Comment by kain88-de** [comment]
aha found the error. When I use numpy version `1.10.4` on conda it fails. A version below it will run smoothly.

**Comment by kain88-de** [comment]
ok so in `transformations.shear_from_matrix` the returned vectors are minimally different. This leasds to that dot-product for `direct` and `normal` calculated from a matrix to have a value around `1e-5`. While older versions of numpy give a result around `1e-16` which is a lot better. I don't know where this is coming from though currently.

**Comment by kain88-de** [comment]
hm `np.linalg.eig` returns different values using different numpy versions. So this doesn't affect us but is likely a failure of the conda-version of numpy=1.10.4.

**Comment by kain88-de** [comment]
Ok the eigenvalues and vectors are different because the standart numpy in conda now links against the mkl. If I use the nomkl version everything works again as expected.

Looking at the tests again. The randomized tests there have been commented and replaced with fixed value tests. According to `git blame` this was done by @richardjgowers. But the commit message doesn't state what used to fail sometimes. @richardjgowers do you remember?

Could it be that our algorithm to extract a normal from the eigenvectors in M33 in `shear_from_matrix` isn't as accurate as we think it is?

**Comment by richardjgowers** [comment]
I don't 100% remember, but it was something like 1 in every 10 builds failed.  If you uncomment it locally you might be able to reproduce it.

**Comment by kain88-de** [comment]
I've used the following script to find out how often it happens that the direction and normal are not orthogonal.

```python
import numpy as np
import MDAnalysis.lib.transformations as t
from six.moves import range

count = 0
total = 10000

for i in range(total):
    angle = (np.random.random() - 0.5) * 4*np.pi
    direct = np.random.random(3) - 0.5
    point = np.random.random(3) - 0.5
    normal = np.cross(direct, np.random.random(3))

    S0 = t.shear_matrix(angle, direct, point, normal)
    angle, direct, point, normal = t.shear_from_matrix(S0)
    dot = np.dot(normal, direct)
    if dot > 1e-6:
        print("not orthogonal dot = {:e}".format(dot))
        count += 1

print("number of fails = {}, % of total trials = {}".format(count, count/float(total)))
```

In about 0.05% this fails independent if I use a numpy version linked against the MKL or normal BLAS/LAPACK. The can go up to ~0.3 which is pretty high.  My interpretation is that no matter what we use (MKL/LAPACK) the algorithm to extract `direct` and `normal` from a shear matrix has edge-cases where it is very inaccurate. It is bad luck that we hit one for the MKL with the numbers tested now. 

@richardjgowers  would you be ok if I just change the numbers for the tests that work for MKL and LAPACK. Additionally I will add a comment to the test explaining that edge-cases exist.

Running this very rarely I get error messages that no two independent eigenvalues could be found for `shear_from_matrix`.

**Comment by richardjgowers** [comment]
Yeah that seems sane.  The numbers seem high for edge cases, but I checked the package where this is copied from and they're still using the same code.

**Comment by kain88-de** [comment]
Maybe the math in the package is sound. But it could also be that we hit an edge-case in eigenvalue decompostion.  It is most likely their because the same matrix results in different eigenvectors. But since we have a similar failure rate independent of the linalg backend I think it is something we have to live with.  I can ask some friends who might know more about this.
